packages/@react-spectrum/dialog/docs/Dialog.mdx

```diff
@@ -43,52 +47,345 @@ import {Heading, Text} from '@react-spectrum/typography';
         <Button variant="cta" onPress={close}>Confirm</Button>
-      </Footer>
+      </ButtonGroup>
     </Dialog>
   )}
 </DialogTrigger>
 ```
 
 ## Content
 
-*If the component has a children prop that accepts any type of content (e.g. `ReactNode`), include this section. Please include a note about how to internationalize the content.*
+A typical Dialog with a title, contents, and action buttons can be created like so:
+
+```tsx example
+<DialogTrigger>
+  <ActionButton>Publish</ActionButton>
+  {(close) => (
+    <Dialog>
+      <Header><Heading>Publish 3 pages</Heading></Header>
+      <Divider />
+      <Content>Confirm publish?</Content>
+      <ButtonGroup>
+        <Button variant="secondary" onPress={close}>Cancel</Button>
+        <Button variant="cta" onPress={close} autoFocus>Confirm</Button>
+      </ButtonGroup>
+    </Dialog>
+  )}
+</DialogTrigger>
+```
+
+A dismissable Dialog forgoes a footer section and its associated action buttons in favor
+of rendering a close button. If your Dialog is dismissable, make sure that you do not include a footer section or
+a ButtonGroup. While the Dialog will hide any elements occupying the footer section, it will not remove them from the DOM, 
+resulting in possible tab order issues.
+
+```tsx example
+<DialogTrigger isDismissable>
+  <ActionButton>Status</ActionButton> 
+  <Dialog>
+    <Header><Heading>Status</Heading></Header>
+    <Divider />
+    <Content>Printer Status: Connected</Content>
+  </Dialog>
+</DialogTrigger>
+```
+
+It is important to note that the `Heading`, `Content`, and `Footer` content elements accept any renderable 
+node, not just strings. This allows you to create Dialogs for more complex workflows, such as including a form
+for the user to fill out or adding confirmation checkboxes.
+
+```tsx example
+<DialogTrigger>
+  <ActionButton>Register</ActionButton> 
+  {(close) => (
+    <Dialog>
+      <Header><Heading>Register for newsletter</Heading></Header>
+      <Divider />
+      <Content>
+        <Form>
+          <TextField label="First Name" placeholder="John" autoFocus />
+          <TextField label="Last Name" placeholder="Smith" />
+          <TextField label="Street Address" placeholder="123 Any Street" />
+          <TextField label="City" placeholder="San Francisco" />
+        </Form>
+      </Content>
+      <Footer>
+        <Checkbox>
+          I want to receive updates for exclusive offers in my area.
+        </Checkbox>
+      </Footer> 
+      <ButtonGroup>
+        <Button variant="secondary" onPress={close}>Cancel</Button>
+        <Button variant="cta" onPress={close}>Register</Button>
+      </ButtonGroup>
+    </Dialog>
+  )}
+</DialogTrigger>
+```
 
 ### Accessibility
 
-*If the component has accessibility needs, include this section.*
+The tab order within the Dialog follows the order of the children provided to the Dialog. 
+This can be leveraged to create a logical flow separate from the reading order on the page.
```

Comment:
I don't think we should encouraging this in the docs. Maybe exclude this example?